src/audio/copier/copier.c

```diff
@@ -715,8 +738,14 @@ static void copier_free(struct comp_dev *dev)
 		rfree(cd->hd);
 	}
 
+	if (dev->ipc_config.type == SOF_COMP_DAI) {
```

Comment:
`else if`. This time `cd->ipc_gtw` is unimportant? Or you could use `switch (dev->ipc_config.type)`

Reply:
config->type = SOF_COMP_HOST;

gtw assigned type as HOST, so for DAI case, there is no possible with ipc_gtw == true.

Reply:
yes, I will change to this style, previously focused on resolve multisink and other CI related errors.